app/test_main.py

```diff
@@ -1 +1,37 @@
-# write your code here
+import pytest
+from unittest.mock import patch
+from typing import Callable
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "expected_url_address, expected_web_connection, expected_result",
+    [
+        pytest.param(
+            True, True, "Accessible",
+            id="Have an Internet connection and a valid URL-address"
+        ),
+        pytest.param(
+            True, False, "Not accessible",
+            id="URL-address is valid, but "
+               "there is no connection to the Internet"
+        ),
+        pytest.param(
+            False, True, "Not accessible",
+            id="There is a connection to the "
+               "Internet, but the address is not valid"
+        )
+    ]
+)
+@patch("app.main.valid_google_url")
+@patch("app.main.has_internet_connection")
```

Comment:
The order of the `@patch` decorators should match the order of the arguments in the test function. Currently, `mocked_url` is the first argument, but `valid_google_url` is patched second. Swap the order of the `@patch` decorators to ensure `mocked_url` corresponds to `valid_google_url` and `mocked_web_connection` to `has_internet_connection`.